python/lsst/obs/decam/decamMapper.py

```diff
@@ -73,11 +89,18 @@ def __init__(self, inputPolicy=None, **kwargs):
                                      DecamMapper._nbit_filter)
 
     def _extractDetectorName(self, dataId):
+        copyId = self._transformId(dataId)
         try:
-            return DecamMapper.detectorNames[dataId['ccdnum']]
+            return DecamMapper.detectorNames[copyId['ccdnum']]
         except KeyError:
             raise RuntimeError("No name found for dataId: %s"%(dataId))
 
+    def _transformId(self, dataId):
+        copyId = dataId.copy()
+        if "ccd" in copyId:
+            copyId.setdefault("ccdnum", copyId["ccd"])
```

Comment:
One possibility is an user gives both `ccd` and `ccdnum` as dataIds. If `ccdnum` and `ccd` are not the same, I'm not sure if we call it an error, but that is definitely confusing.  

I made a copy because `_transformId` in `cameraMapper.py` does.  Honestly I'm confused with the design of the dataId transformation, but hope DM-4539 will clear up confusion.